apisix/plugins/hmac-auth.lua

```diff
@@ -264,18 +310,22 @@ local function get_params(ctx)
         local auth_data = ngx_re.split(auth_string, "#")
         core.log.info("auth_string: ", auth_string, " #auth_data: ",
             #auth_data, " auth_data: ", core.json.delay_encode(auth_data))
-        if #auth_data == 5 and auth_data[1] == "hmac-auth-v1" then
+        if #auth_data == 6 and auth_data[1] == "hmac-auth-v2" then
             ak = auth_data[2]
             signature = auth_data[3]
             algorithm = auth_data[4]
             timestamp = auth_data[5]
+            signed_headers = auth_data[6]
         end
     end
 
     params.access_key = ak
     params.algorithm  = algorithm
     params.signature  = signature
     params.timestamp  = timestamp or 0
+    params.signed_headers = ngx_re.split(signed_headers or "", ";")
```

Comment:
if the `signed_headers` is `nil`, I think `params.signed_headers` is `nil`.

it is an optional field.